rust/operator-binary/src/config.rs

```diff
@@ -621,10 +622,31 @@ pub fn build_nifi_properties(
             "nifi.security.user.oidc.claim.identifying.user".to_string(),
             provider.principal_claim.to_string(),
         );
-        properties.insert(
-            "nifi.security.user.oidc.truststore.strategy".to_string(),
-            "NIFI".to_string(),
-        );
+        if let Some(tls) = &provider.tls.tls {
+            if let TlsVerification::Server(verification) = &tls.verification {
+                if let CaCert::WebPki {} = verification.ca_cert {
+                    properties.insert(
+                        "nifi.security.user.oidc.truststore.strategy".to_string(),
+                        "JDK".to_string(),
+                    );
+                } else {
+                    properties.insert(
+                        "nifi.security.user.oidc.truststore.strategy".to_string(),
+                        "NIFI".to_string(),
+                    );
+                }
+            } else {
+                properties.insert(
+                    "nifi.security.user.oidc.truststore.strategy".to_string(),
+                    "NIFI".to_string(),
+                );
+            }
+        } else {
+            properties.insert(
+                "nifi.security.user.oidc.truststore.strategy".to_string(),
+                "NIFI".to_string(),
+            );
```

Comment:
You can use a match statements instead of multiple `if let`.

Please check the `todo!` marker. This code also does not set `nifi.security.user.oidc.truststore.strategy` when TLS is not being used, but I'd say that's desirable - but happy to argue!

```suggestion

        if let Some(tls) = &provider.tls.tls {
            let truststore_strategy = match tls.verification {
                TlsVerification::None {} => {
                    todo!("Does Nifi really support not checking the cert? If not please raise an error")
                }
                TlsVerification::Server(TlsServerVerification {
                    ca_cert: CaCert::SecretClass(_),
                }) => "NiFi", // The cert get's added to the stackable truststore
                TlsVerification::Server(TlsServerVerification {
                    ca_cert: CaCert::WebPki {},
                }) => "JDK", // The cert needs to be in the system truststore
            };
            properties.insert(
                "nifi.security.user.oidc.truststore.strategy".to_owned(),
                truststore_strategy.to_owned(),
            );
        }
```